personalize.admin.inc

```diff
@@ -915,120 +750,11 @@ function personalize_agent_date_form_submit($agent_name, $values) {
     else {
       $end_date = 0;
     }
-    $varname = _personalize_agent_get_stoptime_variable($agent_name);
-    variable_set($varname, $end_date);
+    personalize_agent_set_stop_date($agent_name, $end_date);
   }
 }
 
 /**
- * Ajax callback for agent form to reset the current subform.
- */
-function personalize_agent_form_ajax_reset($form, &$form_state) {
-  // Clear custom storage.
-  unset($form_state['to_remove']);
-  unset($form_state['num_goals']);
-  // Clear user input.
-  unset($form_state['complete form']);
-  $form_state['input'] = array();
-  $form_state['values'] = array();
-  // Rebuild the form.
-  return drupal_rebuild_form($form['#form_id'], $form_state, $form);
-}
-
-/**
- * Ajax callback for agent edit page form ajax submissions
- */
-function personalize_agent_form_ajax_submit($form, $form_state) {
-  $commands = array();
-  $agent_data = NULL;
-  if (isset($form_state['values']['agent'])) {
-    $agent_data = personalize_agent_load($form_state['values']['agent'], TRUE);
-  }
-
-  if ($form['#form_id'] == 'personalize_agent_form') {
-    // Configuration the agent pauses the agent status so toggle and
-    // option sets must also be updated.
-    $agent_data = personalize_agent_load($form_state['values']['agent_basic_info']['machine_name'], TRUE);
-    $agent_instance = personalize_agent_load_agent($agent_data->machine_name);
-    $option_sets_form = drupal_get_form('personalize_agent_option_sets_form', $agent_data, $agent_instance);
-    $commands[] = ajax_command_replace('#personalize-agent-option-sets-form', drupal_render($option_sets_form));
-    // This could be triggered by the status toggle so enforce reloading the
-    // full personalize-agent-form regardless of the submit used.
-    $form_div_id = 'personalize-agent-form';
-  }
-  else if ($form['#form_id'] == 'personalize_agent_option_sets_form') {
-    // Updating the option sets causes the campaign to pause so we need to
-    // update display of the status toggle.
-    $campaign_form = drupal_get_form('personalize_agent_form', $agent_data);
-    $commands[] = ajax_command_replace('#personalize-agent-form', drupal_render($campaign_form));
-  }
-
-  // Update the current agent status.
-  if (!empty($agent_data)) {
-    $current_status = personalize_agent_get_status($agent_data->machine_name);
-    list($next_status, $status_text) = _personalize_status_toggle_next($current_status);
-    $settings = array(
-      'personalize' => array(
-        'campaigns' => array(
-          $agent_data->machine_name => array(
-            'status' => $current_status,
-            'nextStatus' => array(
-              'status' => $next_status,
-              'text' => $status_text,
-            ),
-          ),
-        ),
-      ),
-    );
```

Comment:
Yay :) (Although this also means your branch isn't based off the latest RC - I think you have some fun rebasing in your future :/)